src/prng/chacha.rs

```diff
@@ -10,13 +10,10 @@
 
 //! The ChaCha random number generator.
 
-use core::num::Wrapping as w;
 use core::fmt;
+use rand_core::impls;
 use {Rng, CryptoRng, SeedFromRng, SeedableRng, Error};
 
-#[allow(bad_style)]
-type w32 = w<u32>;
-
```

Comment:
I'm not sure that removing this is actually a win... I mean now you have `.wrapping_add` in a few places and can't just think *I know this algorithm uses wrapping arithmetic everywhere*.

Reply:
In the commit message for ChaCha I added this note:

> This also replaces `core::num::Wrapping` with a few `wrapping_add`'s.
> There were about 30 conversions to and from `Wrapping`, while there are only
> 9 wrapping operations.
>
> Because `fill_via_u32_chunks` expects a `[u32]`, converting away was just
> easier.

I agree that _I know this algorithm uses wrapping arithmetic everywhere_ is an advantage. Not all operations are available on wrapping types though, like `rotate_*`. You can maybe consider this to be a bug in the standard library.

While working with ISAAC, XorShift* and PCG it happened to many times I had to ask myself if I was working with the wrapped or the normal type, and if an operation was available.